osu.Desktop/Performance/HighPerformanceSessionManager.cs

```diff
@@ -11,16 +12,26 @@ namespace osu.Desktop.Performance
 {
     public class HighPerformanceSessionManager : IHighPerformanceSessionManager
     {
+        public bool IsSessionActive => activeSessions > 0;
+
+        private int activeSessions;
+
         private GCLatencyMode originalGCMode;
 
         public IDisposable BeginSession()
         {
-            enableHighPerformanceSession();
-            return new InvokeOnDisposal<HighPerformanceSessionManager>(this, static m => m.disableHighPerformanceSession());
+            enterSession();
+            return new InvokeOnDisposal<HighPerformanceSessionManager>(this, static m => m.exitSession());
         }
 
-        private void enableHighPerformanceSession()
+        private void enterSession()
         {
+            if (Interlocked.Increment(ref activeSessions) > 1)
+            {
+                Logger.Log($"High performance session requested ({activeSessions} others already running)");
```

Comment:
Nit, but this is already post-increment, so you'd either want

```suggestion
                Logger.Log($"High performance session requested ({activeSessions - 1} others already running)");
```

or just

```suggestion
                Logger.Log($"High performance session requested ({activeSessions} running in total)");
```